src/modules/esl-panel-group/core/esl-panel-group.ts

```diff
@@ -283,12 +283,13 @@ export class ESLPanelGroup extends ESLBaseElement {
 
     const max = this.currentMaxItems;
     const params = this.mergeActionParams({event: e});
+    const balanceMarker = e.detail?.params?.event?.type !== 'esl:before:hide';
 
     // All currently active except panel that requested to be open
     const $activePanels = this.$activePanels.filter((el) => el !== panel);
 
-    // overflow = pretended to be active (current active + requested panel) - limit
-    const overflow = Math.max(0, $activePanels.length + 1 - max);
+    // overflow = pretended to be active (current active + balanceMarker (1 if nothing hides)) - limit
+    const overflow = Math.max(0, $activePanels.length + Number(balanceMarker) - max);
```

Comment:
```suggestion
    const overflow = Math.max(0, $activePanels.length + (balanceMarker ? 1 : 0) - max);
```
the ternary solution works faster.

Also you can modify line 281 to
`const {target: panel, detail} = e;`
and 286 line to
`const balanceMarker = detail?.params?.event?.type !== 'esl:before:hide';`

Reply:
"the ternary solution works faster" 
Nobody will see benefits from that, even if it's faster, its less than 0.01ms
And even to test that that's correct, you need to spend a lot of effort (a lot of tests is absolutely lie due to V8 optimizations for routines that happen in "synthetics" tests) ;)

So disagree for that change as a strict rule for our code style.